frontend/src/metabase/actions/containers/ActionCreator/ActionCreator.tsx

```diff
@@ -175,6 +181,14 @@ function ActionCreator({
   );
 }
 
+function maybeCloseAceEditor() {
+  const ace = "ace" in window ? (window.ace as any) : null;
+  const editor = ace?.edit(ACE_ELEMENT_ID);
```

Comment:
I swear to god I have a comment for this line, but it disappeared or maybe I just didn't submit the comment 🤪 .

Another way to solve this is to expose a function to close an Ace editor as opposed to exposing the ID itself. I think that way, the logic for dealing with Ace editor is at least more self-contained. What do you think?

Reply:
Do you mean adding a method closing the popover to `NativeQueryEditor`? If so, I believe it doesn't worth the hassle of passing down react refs. There's a pretty high chance the editor is going to change to avoid nested models (action editor being a modal + having the save modal on top)

Reply:
Can't we just put this very same function in `NativeQueryEditor`? It can use `ACE_ELEMENT_ID` to query the element. We just change the location to house this logic.

Reply:
I'd keep it here: otherwise, there's a pretty high chance we'll forget to delete it when we don't need it anymore